### PR TITLE
Add WithErrorLogs interceptor

### PR DESCRIPTION
**Pull request by sp-francisco-moreno**

### CODE REVIEW

File: grpcx/server.go [hunk 1]
```diff
@@ -53,3 +53,26 @@ func WithRequestResponseLogs(l logx.Logger) grpc.UnaryServerInterceptor {
...
+// WithErrorLogs returns a gRPC interceptor for unary calls that instrument requests
+// with logs for the errors.
+func WithErrorLogs(l logx.Logger) grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		resp, err := handler(ctx, req)
+
+		if err != nil {
+			reqMsg, _ := json.Marshal(req)
+			respMsg, _ := json.Marshal(resp)
+			fields := []logx.Field{
```
Comment (sp-eric-pinto) [inline grpcx/server.go:66]: no variable is required, use it directly

File: grpcx/server.go [hunk 2]
```diff
@@ -53,3 +53,26 @@ func WithRequestResponseLogs(l logx.Logger) grpc.UnaryServerInterceptor {
...
+				{Key: "url", Value: info.FullMethod},
+				{Key: "ctx_full_method", Value: info.FullMethod},
```
Comment (sp-eric-pinto) [inline grpcx/server.go:68]: remove this line, "url" key is fine

**Review by sp-eric-pinto — CHANGES_REQUESTED** [review]

File: grpcx/server.go (hunk 2 above)
Comment (sp-francisco-moreno) [reply grpcx/server.go:69]: As commented on the channel, I'm going to log on some redundant fields, for the shake of visibility on multiple DD panels.

**Review by sp-eric-pinto — APPROVED** [review]

**Review by sp-manuel-peralta — APPROVED** [review]

File: grpcx/server_test.go [hunk 3]
```diff
@@ -110,3 +110,71 @@ func TestWithRequestResponseLogs(t *testing.T) {
...
+	t.Run("logs error", func(t *testing.T) {
+		w := bytes.NewBufferString("")
+		l := logx.New(logx.WriterOpt(w))
+
+		ctx := context.Background()
+		req := struct {
+			UserID string
+		}{
+			UserID: userID,
+		}
```
Comment (sp-jordi-forns) [inline grpcx/server_test.go:158]: you could move these repeated lines out of the func and share it between the two tests, to save some space.

**Review by sp-jordi-forns — APPROVED** [review]

File: grpcx/server_test.go (hunk 3 above)
Comment (sp-francisco-moreno) [reply grpcx/server_test.go]: done :)

**Pull request merged**